extensions-core/druid-bloom-filter/src/main/java/org/apache/druid/query/filter/sql/BloomFilterOperatorConversion.java

```diff
@@ -100,7 +100,8 @@ public DimFilter toDruidFilter(
       return new BloomDimFilter(
           druidExpression.getSimpleExtraction().getColumn(),
           holder,
-          druidExpression.getSimpleExtraction().getExtractionFn()
+          druidExpression.getSimpleExtraction().getExtractionFn(),
```

Comment:
Yeah, this is true, but I was trying to avoid using these constructors in production code to minimize the chance of errors and make things more obvious.

I initially had it more like you are suggesting and was using some of the 'test' constructors where we were always passing null, but [this comment](https://github.com/apache/incubator-druid/pull/8209#discussion_r310319858) suggested the way it is now and I agree, I think the production code explicitly passing in `null` for the `filterTuning` parameter makes it less ambiguous about if not having a tuning is intentional or not, without requiring a comment.

Reply:
I think it's anyway strange that there is a chaining constructor and any code is not using an existing constructor with fewer parameters.

I think it would be better to create static factory methods instead in this situation. Some of them may have "InTest" suffix to strongly indicate that they are not for prod code.

Reply:
I agree, but the reason I have a second set of constructors at all was so I didn't have to change hundreds of lines of test code to add a `null` parameter, so making a static method would sort of defeat the purpose of that because the lines would change anyway.

I do think this is worth doing, I'll add this to the ticket i create about refactoring `DimFilter` to also add useful test filter creation static methods, so we can also eliminate passing in `null` everywhere for filters without `extractionFn`.

Reply:
mentioned making static test constructors to decouple test filters from json schema in #8256